src/inference/include/openvino/runtime/async_infer_queue.hpp

```diff
@@ -0,0 +1,197 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file that provides wrapper classes for infer requests and callbacks.
+ *
+ * @file openvino/runtime/async_infer_queue.hpp
+ */
+#pragma once
+
+#include <condition_variable>
+#include <map>
+#include <mutex>
+#include <queue>
+#include <string>
+#include <vector>
+
+#include "ie/ie_plugin_config.hpp"
+#include "openvino/runtime/common.hpp"
+#include "openvino/runtime/executable_network.hpp"
+#include "openvino/runtime/infer_request.hpp"
+#include "openvino/runtime/tensor.hpp"
+
+namespace helpers {
+/**
+ * @brief Helper function to get number of jobs based on given network's metrics.
+ *
+ * @param net ExecutableNetwork object.
+ *
+ * @return Returns number of jobs.
+ */
+static size_t num_of_jobs_helper(ov::runtime::ExecutableNetwork& net) {
```

Comment:
This function is used outside of this file so I suggest to move the implementation to the .cpp file.

Reply:
Resolved offline.